tests/src/test/java/alluxio/client/cli/job/DistributedCmdTest.java

```diff
@@ -0,0 +1,182 @@
+            MetricKey.MASTER_JOB_DISTRIBUTED_LOAD_FILE_SIZE.getName()).getValue();
+    double migrateFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_MIGRATE_JOB_FILE_SIZE.getName()).getValue();
+    double persistFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_ASYNC_PERSIST_FILE_SIZE.getName()).getValue();
+
+    assertEquals(loadFileSize, LOAD_FILE_LENGTH, 0); // should equal LOAD_FILE_LENGTH
+    assertEquals(migrateFileSize, DEFAULT_ZERO_LENGTH, 0); // should equal DEFAULT_ZERO_LENGTH
+    assertEquals(persistFileSize, DEFAULT_ZERO_LENGTH, 0); // should equal DEFAULT_ZERO_LENGTH
+  }
+
+  @Test
+  public void testMigrateConfigForCompleteStatus() throws Exception {
+    // Run for migrate config.
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
+            mMigrateJobConfig, mFileSystem, mRetryPolicy);
+
+    // should expect function calls to fileSystem.getStatus(new AlluxioURI(filePath)).getLength()
+    // for MigrateConfig only and cooresponding file sizes obtained.
+    double loadFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_JOB_DISTRIBUTED_LOAD_FILE_SIZE.getName()).getValue();
+    double migrateFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_MIGRATE_JOB_FILE_SIZE.getName()).getValue();
+    double persistFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_ASYNC_PERSIST_FILE_SIZE.getName()).getValue();
+
+    assertEquals(loadFileSize, DEFAULT_ZERO_LENGTH, 0); // should equal DEFAULT_ZERO_LENGTH
+    assertEquals(migrateFileSize, MIGRATE_FILE_LENGTH, 0); // should equal MIGRATE_FILE_LENGTH
+    assertEquals(persistFileSize, DEFAULT_ZERO_LENGTH, 0); // should equal DEFAULT_ZERO_LENGTH
+  }
+
+  @Test
+  public void testPersistConfigForCompleteStatus() throws Exception {
+    // Run for persist config.
+    DistributedCmdMetrics.incrementForCompleteStatusWithRetry(
+            mPersistJobConfig, mFileSystem, mRetryPolicy);
+
+    // should expect function calls to fileSystem.getStatus(new AlluxioURI(filePath)).getLength()
+    // for PersistConfig only and cooresponding file sizes obtained.
+    double loadFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_JOB_DISTRIBUTED_LOAD_FILE_SIZE.getName()).getValue();
+    double migrateFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_MIGRATE_JOB_FILE_SIZE.getName()).getValue();
+    double persistFileSize = MetricsSystem.getMetricValue(
+            MetricKey.MASTER_ASYNC_PERSIST_FILE_SIZE.getName()).getValue();
+
+    assertEquals(loadFileSize, DEFAULT_ZERO_LENGTH, 0); // should equal DEFAULT_ZERO_LENGTH
+    assertEquals(migrateFileSize, DEFAULT_ZERO_LENGTH, 0); // should equal DEFAULT_ZERO_LENGTH
+    assertEquals(persistFileSize, PERSIST_FILE_LENGTH, 0); // should equal PERSIST_FILE_LENGTH
+  }
+
+  @Test
+  public void testUnSupportedConfigForCompleteStatus() throws Exception {
```

Comment:
I think this still doesn't cover the failure we are seeing which we call get URI couple times for null? We are testing with NoOp config the supported metrics don't change.

Reply:
I think we are testing for unsupported configs, the call to filesystem.getStatus doesn't get called. Here i just used an unsupported NoopPlan config which does not belong to Load, Migrate or Persist types. Do I miss anything?

Reply:
I think even in the old implementation this test would still pass, isn't it? The purpose of this test is that we should fail the old implementation.

Reply:
discussed offline and updated PR by adding verify function call directly.